File: implementation/scipamato/core/logic/src/test/java/ch/difty/scipamato/core/logic/parsing/DefaultAuthorParserFactoryTest.java
package ch.difty.scipamato.core.logic.parsing;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import ch.difty.scipamato.common.NullArgumentException;

class DefaultAuthorParserFactoryTest {

    private AuthorParserFactory factory;

    @BeforeEach
    void setUp() {
        factory = new DefaultAuthorParserFactory(AuthorParserStrategy.PUBMED);
    }

    @Test
    void degenerateConstruction() {
        Assertions.assertThrows(NullArgumentException.class, () -> new DefaultAuthorParserFactory(null));
    }

    @Test
    void creatingParser_withNullAuthorString_throws() {
        Assertions.assertThrows(NullArgumentException.class, () -> factory.createParser(null));
    }

    @Test
    void cratingParser_withNoSetting_usesDefaultAuthorParser() {
        AuthorParser parser = factory.createParser("Turner MC.");
        assertThat(parser).isInstanceOf(PubmedAuthorParser.class);
    }

}

File: implementation/scipamato/core/logic/src/test/java/ch/difty/scipamato/core/logic/parsing/DefaultAuthorParserFactoryTest.kt
package ch.difty.scipamato.core.logic.parsing

import ch.difty.scipamato.common.NullArgumentException
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class DefaultAuthorParserFactoryTest {

    private lateinit var factory: AuthorParserFactory

    @BeforeEach
    fun setUp() {
        factory = DefaultAuthorParserFactory(AuthorParserStrategy.PUBMED)
    }

    @Test
    fun degenerateConstruction() {
        Assertions.assertThrows(NullArgumentException::class.java) { DefaultAuthorParserFactory(null) }
    }

    @Test
    fun creatingParser_withNullAuthorString_throws() {
        Assertions.assertThrows(NullArgumentException::class.java) { factory.createParser(null) }
    }

    @Test
    fun cratingParser_withNoSetting_usesDefaultAuthorParser() {
        val parser = factory.createParser("Turner MC.")
        assertThat(parser).isInstanceOf(PubmedAuthorParser::class.java)
    }

}